airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals;
+
+import com.google.common.collect.AbstractIterator;
+import io.airbyte.integrations.debezium.CdcMetadataInjector;
+import io.airbyte.integrations.debezium.CdcStateHandler;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.debezium.engine.ChangeEvent;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class encapsulates an iterator and adds the required functionality to create checkpoints for
+ * CDC replications. That way, if the process fails in the middle of a long sync, the process is
+ * able to recover for any acknowledged checkpoint in the following syncs.
+ */
+public class DebeziumStateDecoratingIterator extends AbstractIterator<AirbyteMessage> implements Iterator<AirbyteMessage> {
+
+  public static final Integer SYNC_CHECKPOINT_SECONDS = 15 * 60;
```

Comment:
Can we mark it as a Duration? IT makes things simpler to understand.
`public static final Duration SYNC_CHECKPOINT_SECONDS = Duration.ofMinutes(15);`

Reply:
Ok, I had changed also how we set up the value as well, so then we fit all parameters with `Duration` object type:
```
    final Duration syncCheckpointSeconds = config.get("sync_checkpoint_seconds") != null ? Duration.ofSeconds(config.get("sync_checkpoint_seconds").asInt())
        : DebeziumStateDecoratingIterator.SYNC_CHECKPOINT_SECONDS;
```